src/client/client.go

```diff
@@ -47,6 +47,8 @@ type APIClient struct {
 var (
 	//BlockSize is used internally by PFS blockserver
 	BlockSize = 8 * 1024 * 1024 // 8 Megabytes
+	// MaxBlockSize specifies the maximum block size for any data type
+	MaxBlockSize = 100 * 1024 * 1024 // 100 MB
 	// MaxMsgSize is used to define the GRPC frame size, which we need to be greater than a block
 	MaxMsgSize = 3 * BlockSize
```

Comment:
So before that buffer change, we realized that in some cases we were writing a whole block at once. This had to do with the fact that the implementation of `io.Copy` uses [some tricks](https://golang.org/src/io/io.go?s=12490:12550#L379) and doesn't actually copy under the hood if it doesn't have to.

So in the case of test cases where you were using a `bytes.NewReader()` for instance, `io.Copy` would do a single `write()` call w the whole buffer. This amounts to writing the whole block. If the reader was an `os.File` I believe it was written in smaller chunks. But it depended on the reader, not the writer. So the buffered write fix enforces that we write in small chunks all the time now.

Hopefully that clears that part up?

So before this buffered write change, a single block could be written all at once. (At least in a common way we were writing tests) This is why `blockSize` was conflated with `MaxMsgSize`. I agree they shouldn't be. And with the code we have today, they are no longer functionally coupled at all.

[I've created an issue to decouple these values](https://github.com/pachyderm/pachyderm/issues/1335)